### DISCUSSION TITLE
q-input type prop mere documentation or triggering automated validation?

### DISCUSSION THREAD

**Discussion by JacquesRobin** [quasarframework/quasar#14240] [General - Components / Directives / etc]
Newbie here who couldn't easily find the answer to his question:
Is the type prop of a q-input component a mere documentation item, or does it trigger built-in validation rules checking that the input content indeed conforms to the type declared for the prop?

For example, a well-formed email input must contain at the very least:
- a non-empty string
- followed by a "@" character
- followed by a non-empty string
- followed by a "." character
- followed by a non-empty string.

So if I defined a q-input component of type email , will it contain out-of-the-box the validation rules checking the constraints listed above, or do I need to write the validation rules myself?

Thanks!

**Comment by pdanpdan** [comment]
The rules will be the one implemented by the browser, and the validation will be at the level of the html input, not the Vue component
